###Coding Discussion 2

In [2]:
#load packages
import csv

#read the data
with open('us-states.csv') as file:
    state_covid_data = []
    for row in csv.reader(file):
        state_covid_data.append(row)

##Count up the number of unique dates in the data.

In [3]:
#1
"""
This function takes a nested list of strings and searches through each sublist for strings containing a specific term and creates a list of unique occurences of strings that term appears in.

Arguments
---------
list: list
    a nested list of strings
term: str
    a string of text, special charachters, or numbers
    
return
-------
the length of the list of unique occurences
"""
def unique_dates(list=[], term=''):
    list_of_dates = [] #create an empty list
    for index in range(len(list)): #iterate through the first layer, each sublist
        for index_2 in range(len(list[index])): #iterate through the second layer, each sublist's contents
            if term in list[index][index_2]: #find a character in contents of sublist
                list_of_dates.append(list[index][index_2]) #add each date to the list_of_dates list
    unique_list_of_dates = set(list_of_dates) #apply set function to return a list_of_unique_dates
    return len(unique_list_of_dates) #return the desired value

In [4]:
#1 
unique_dates(state_covid_data,'-')

225

There are 225 unique dates.

##Write a function that takes in a state name as input (e.g. "Wisconsin") and outputs the date of its first case.

In [5]:
#answer to #3
"""
This function takes a nested list of strings and searches through each sublist for strings containing a specified state, creates a list of the sublists containing said state and returns the date associated with the first observation containing said state.
Argument
---------
list: list
    a nested list of strings.
state_name: str
    a string of text referring to the name of a state.
value: int
    an integer value referring to the index of the sublist
return
-------
the date associated the first occurence of that state in the data or a message marking clear that the state had no observations.
"""
def first_occurence(list=[], state_name=''):
    list_of_occurences = [] #create an empty list
    for index in range(1,len(list)): #iterate through the first layer, each sublist
        for index_2 in range(len(list[index])): #iterate through the second layer, each sublist's contents
            if state_name.lower() == list[index][index_2].lower(): #find the state string in contents of sublist.    
                list_of_occurences.append(list[index]) #add the entire sublist to the list of occurences
    if len(list_of_occurences) != 0:
        return list_of_occurences[0][0] #return the desired value for the first occurence of the state in the data.
        
    else:
            return "There are no observations for that state."

##Find the first date in which the District of Columbia recorded a case.

In [6]:
first_occurence(state_covid_data, "District of Columbia")

'2020-03-07'

The first recorded case in DC was March 7, 2020.

##Write a function that takes in a state name as input (e.g. "Wisconsin") and outputs the date when the number of reported cases within the state exceeded 1000.

In [ ]:
"""
This function takes a nested list of strings and searches through each sublist and returns a date for the sublist containing a specified state and that has a number of cases greater than or equal the requested value.
Argument
---------
list: list
    a nested list of strings.
state_name: str
    a string of text referring to the name of a state.
caase_number: int
    an integer value for the number of cases you are interested in finding.
return
-------
the date associated the first occurence of that state where the cases value is greater than case_number.
"""   
def first_day_x_cases_or_more(list=[],state_name='',case_number=int):
    list_of_occurences = [] #create an empty list
    for index in range(1,len(list)): #iterate through the first layer, each sublist
        for index_2 in range(len(list[index])): #iterate through the second layer, each sublist's contents
            if (int(list[index][3]) > case_number):
                if state_name.lower() == list[index][index_2].lower(): #find the state string in contents of sublist.
                    list_of_occurences.append(state_covid_data[index]) #add the entire sublist to the list of occurences
    if len(list_of_occurences) != 0:
        return list_of_occurences[0][0] #return the desired value for the first occurence of the state in the data.
        print(list_of_occurences)
    else:
            return "There are no instances of that state exceeding that many cases."